tests/framework/web/RequestTest.php

```diff
@@ -146,4 +147,35 @@ public function testResolve()
         $this->assertEquals(['post/view', ['id' => 21, 'token' => 'secret']], $result);
         $this->assertEquals($_GET, ['id' => 63]);
     }
+
+    public function testResolveSuffixRedirect()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'urlManager' => [
+                    'enablePrettyUrl' => true,
+                    'showScriptName' => false,
+                    'cache' => null,
+                    'suffix' => '/',
+                    'rules' => [
+                        'posts' => 'post/list',
+                    ],
+                ]
+            ]
+        ]);
+
+        $request = new Request();
+        $request->hostInfo = 'http://example.com/';
+        $request->pathInfo = 'posts/list';
+        $request->baseUrl = '';
+
+        try {
+            $request->resolve();
+        } catch (RedirectException $exception) {
+            // catch redirect exception
+        }
+        $this->assertTrue(isset($exception));
+        $this->assertEquals('/posts/list/', $exception->url);
```

Comment:
Overal set of tests to add
1) test redirection of:
- example.com/path/controller/action ->  /path/controller/action/
- example.com/path/controller/action ->  /path/controller/action/
- example.com/path/index.php/controller/action ->  /path/index.php/controller/action/
- example.com/path/index.php/module/controller/action ->  /path/index.php/module/controller/action/

2) the other way around
- example.com/path/controller/action/ ->  /path/controller/action
- etc

3) UrlRules
- a hard rules setup (like accepting both /contact and /contact/ )
- a UrlRule suffix scenario